.github/workflows/lint-backend.yml

```diff
@@ -1,6 +1,9 @@
 name: Python Linting with Ruff
 
-on: push
+on:
+  push:
+    branches: [ "main" ]
+  pull_request:
```

Comment:
The reason why I did not add `branches` options for `pull_request` is that I want to trigger the workflows even when creating a pull request towards someone's feature branch.

On the other hand, triggering workflows for every push might overuse the resources. If I create 10 feature branches and push 10 commits for each one by one, 100 CI runs will be triggered.